### ISSUE TITLE
Help wanted 🤲 Qwen Training stuck for >20min, GPU memory not released until WSL shutdown

### ISSUE THREAD

**Issue by kamel-msaoubi** [kohya-ss/musubi-tuner#475]
### Describe the bug
When launching a QWEN training with accelerate launch, the process gets stuck indefinitely after model loading (see logs below).

Nothing progresses for 20+ minutes.

Killing the process takes an unusually long time.

GPU memory is not freed unless I perform wsl --shutdown.

This prevents me from running any training.

### To Reproduce
Command I ran:

accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 src/musubi_tuner/qwen_image_train_network.py \
  --dit "/home/ubuntu/models/qwen_image_bf16.safetensors" \
  --vae "/home/ubuntu/models/diffusion_pytorch_model.safetensors" \
  --text_encoder "/home/ubuntu/models/qwen_2.5_vl_7b.safetensors" \
  --dataset_config "/mnt/h/trained-models/trainings/6/config_dataset.toml" \
  --sdpa --mixed_precision bf16 \
  --timestep_sampling shift \
  --discrete_flow_shift 3.0 \
  --optimizer_type adamw8bit --learning_rate 1e-4 --gradient_checkpointing \
  --max_data_loader_n_workers 2 --persistent_data_loader_workers \
  --network_module networks.lora_qwen_image \
  --network_dim 32 \
  --max_train_epochs 16 --save_every_n_epochs 1 --seed 42 \
  --output_dir "//home/ubuntu/trained-models/trainings/6/model" --output_name name-of-lora \
  --blocks_to_swap 45 --fp8_base --fp8_scaled \
  --sample_every_n_epochs 4 --sample_prompts "/mnt/h/trained-models/trainings/6/model/sample/prompt.txt"

Logs

<details> <summary>Full log output</summary>
The following values were not passed to `accelerate launch` and had defaults used instead:
        `--num_processes` was set to a value of `1`
        `--num_machines` was set to a value of `1`
        `--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
Trying to import sageattention
Successfully imported sageattention
INFO:musubi_tuner.hv_train_network:Load dataset config from /mnt/h/trained-models/trainings/6/config_dataset.toml
INFO:musubi_tuner.dataset.image_video_dataset:glob images in /home/ubuntu/trainings/6/img/1_diadog
INFO:musubi_tuner.dataset.image_video_dataset:found 5 images
...
INFO:musubi_tuner.hv_train_network:use 8-bit AdamW optimizer | {}
override steps. steps for 16 epochs is / 指定エポックまでのステップ数: 80

After this line, nothing happens for more than 20 minutes.

</details>

### Hardware/Software stack

GPU: NVIDIA RTX 5070 Ti (16GB)

Driver: CUDA 12.9

OS: Windows 11 with WSL2 (Ubuntu 20.04)

Python: 3.10 (conda env)

Intel i9 12th generation 3.20Ghz
56Go RAM

Frameworks:

- wsl -l -v

  NAME                   STATE           VERSION
  docker-desktop-data    Running         2
  Ubuntu-20.04           Running         2
  docker-desktop         Running         2
  
- nvidia-smi
Tue Aug 19 16:40:40 2025
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 576.88                 Driver Version: 576.88         CUDA Version: 12.9     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                  Driver-Model | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 5070 Ti   WDDM  |   00000000:01:00.0 Off |                  N/A |
| 32%   52C    P8             12W /  300W |     229MiB /  16303MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A            1488    C+G   C:\Windows\System32\dwm.exe           N/A      |
|    0   N/A  N/A           25212      C   ...mfy\python_embeded\python.exe      N/A      |
+-----------------------------------------------------------------------------------------+

torch 2.8.0+cu129
torchvision 0.23.0+cu129
accelerate 1.6.0

bitsandbytes (installed but without GPU support, warning raised)

### Expected behavior
Training should start after dataset preparation and optimizer initialization.

### Additional context

bitsandbytes seems to fall back to CPU-only mode, even though CUDA 12.9 is installed.

Unsure if the hang is related to LoRA + FP8 swap blocks combination or adamw8bit.

Releasing GPU memory requires a full WSL shutdown.

My wslconfig:
`[wsl2]

memory=16GB

swap=128GB

swapfile=H:\\DockerData\\wsl-swap.vhdx`

 ### 💡Help Wanted:
Has anyone successfully trained with similar config (Qwen2.5-VL + bf16 + LoRA + FP8 swap) on WSL2?
Any workaround to prevent the process from hanging and blocking GPU memory?

**Comment by kamel-msaoubi** [comment]
I was trying to make it work with WSL2 because I supposed torch with cuda129 was not prepared for windows.
Then I came to this issue

#447 

I will try to install exactly same packages and retry a training

**Comment by kohya-ss** [comment]
Since `--blocks_to_swap` option consumes main RAM, 16GB of memory allocated to WSL may be too small. Considering the temporary memory consumption during loading, I recommend 32-48GB.
